File: src/Client/Logistics.OfficeApp/src/app/core/api/models/payment/payment.dto.ts
import {PaymentFor, PaymentMethodType, PaymentStatus} from "@/core/enums";
import {AddressDto} from "../address.dto";

export interface PaymentDto {
  id: string;
  createdDate: string;
  paymentDate?: string;
  method?: PaymentMethodType;
  amount: number;
  status: PaymentStatus;
  paymentFor: PaymentFor;
  billingAddress?: AddressDto;
  comment?: string;
}

File: src/Shared/Logistics.Shared.Models/Payment/PaymentDto.cs
using Logistics.Domain.Primitives.Enums;
using Logistics.Domain.Primitives.ValueObjects;

namespace Logistics.Shared.Models;

public record PaymentDto
{
    public Guid Id { get; set; }
    public DateTime CreatedDate { get; set; }

    public Money Amount { get; set; } = null!;
    public string? StripePaymentMethodId { get; set; }
    public Guid TenantId { get; set; }
    public PaymentStatus Status { get; set; }
    public Address? BillingAddress { get; set; }
    public string? Description { get; set; }
}
